guardian/typerighter
# Allow Composer to filter out dictionary matches

## What does this change?

This PR allows Composer to request a list of matcher categories that it wants included in the checker response. This is the mechanism by which we can, for example, ask for all categories except dictionary rules, in order to exclude them in Composer (if the user toggles them off).

It does this by: 
- Making sure that our `DictionaryRule`s have the category 'Collins Dictionary' by setting the `CATEGORY` in `MorfologikCollinsSpellerRule`.
-  Changing `getApplicableCategories` to return all categories when it is given an empty Set. Until now, an empty list of categories passed to the `checkStream` endpoint gave an empty `Some(Set.empty)` rather than `None` - which meant `getApplicableCategories` would say there were no applicable categories. Basically, if we pass an empty set of categories from this branch, it will consider all categories valid. 

It also stops dictionary rules with multiple suggestions coming through as orange 'Review' matches when they should be be red 'Amend' matches.

## How to test

Test with https://github.com/guardian/flexible-content/pull/4391/files, which will require you to use https://github.com/guardian/prosemirror-typerighter/pull/257 locally.

However, this PR should be good to merge independently of the other two, as long as the last one merged is the flexible-content PR.

## How can we measure success?

- Requesting all categories except dictionary rules returns the expected result
- Request an empty list of categories returns results from all matchers
